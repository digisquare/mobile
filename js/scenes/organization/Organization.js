import React, { Component, PropTypes } from 'react';
import { View, Text, Image, ScrollView, StyleSheet, Linking, TouchableOpacity } from 'react-native';
import { connect } from 'react-redux';
import Icon from 'react-native-vector-icons/FontAwesome';

import HTMLView from 'react-native-htmlview';

import socialBadge from '../../utils/social';

import DigiColors from '../../common/DigiColors';
import VenueFooter from '../venues/VenueFooter';

const styles = StyleSheet.create({
  itemWrapper: {
    flex: 1,
    justifyContent: 'center',
    padding: 10,
  },
  container: {
    flex: 1,
    backgroundColor: DigiColors.primaryBackgroundColor,
  },
  scrollview: {
    flex: 1,
  },
  organization: {
    padding: 10,
  },
  header: {
    flex: 1,
    flexDirection: 'row',
    marginBottom: 20,
  },
  avatarContainer: {
    marginRight: 20,
  },
  avatar: {
    width: 80,
    height: 80,
  },
  title: {
    fontSize: 22,
    fontWeight: '900',
    color: DigiColors.primaryFontColor,
    marginTop: 10,
  },
  social: {
    flex: 1,
    flexDirection: 'row',
    marginTop: 10,
  },
});

class Organization extends Component {
  static route = {
    navigationBar: {
      title: 'Organisateur',
      renderRight: (route) => {
        const { params: { organization } } = route;
        return organization.Organization.Contacts.website && (
          <TouchableOpacity
            onPress={() => Linking.openURL(organization.Organization.Contacts.website)}
            style={styles.itemWrapper}
          >
            <Icon name="link" size={20} color={DigiColors.invertedFontColor} />
          </TouchableOpacity>
        );
      },
    },
  }

  render() {
    const { organization, organizations } = this.props;
    const { edition_id: editionId, venue_id: venueId } = organization.Organization;

    if (editionId && organizations[editionId]) {
      const organizationsItems = organizations[editionId].items;
      if (venueId && !organization.Venue && organizationsItems.length > 0) {
        const venue = organizationsItems.find(item => item.Venue.id === venueId);
        organization.Venue = venue ? venue.Venue : null; // eslint-disable-line no-param-reassign
      }
    }

    return (
      <View style={styles.container}>
        <ScrollView style={styles.scrollview}>
          <View style={styles.organization}>
            <View style={styles.header}>
              {
                organization.Organization.avatar ? (
                  <View style={styles.avatarContainer}>
                    <Image
                      style={styles.avatar}
                      source={{ uri: organization.Organization.avatar }}
                    />
                  </View>
                ) : null
              }
              <View style={styles.titleContainer}>
                <Text style={styles.title}>
                  {organization.Organization.name}
                </Text>
                <View style={styles.social}>
                  {socialBadge('twitter', organization.Organization.Contacts)}
                  {socialBadge('facebook', organization.Organization.Contacts)}
                </View>
              </View>
            </View>
            <HTMLView
              value={organization.Organization.description}
              onLinkPress={url => Linking.openURL(url)}
            />
          </View>
        </ScrollView>
        {
          organization.Venue && organization.Venue.id && <VenueFooter venue={organization.Venue} />
        }
      </View>
    );
  }
}

const mapStateToProps = state => ({
  organizations: state.organizations,
});

export default connect(mapStateToProps)(Organization);

Organization.propTypes = {
  organization: PropTypes.object.isRequired,
  organizations: PropTypes.object.isRequired,
};
