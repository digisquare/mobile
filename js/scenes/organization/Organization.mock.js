const Organization = {
  Organization: {
    id: '22',
    venue_id: null,
    edition_id: '9',
    name: 'BordeauxJS',
    type: '0',
    avatar: 'https://pbs.twimg.com/profile_images/2741451195/8ff14ab4849d749af06271795a15cf54_400x400.png',
    description: 'User group JavaScript \u00e0 #bordeaux : r\u00e9unions, discussions et live coding entre gens en asynchrone. #nodejs #coffeescript #javascript',
    contacts: '{\'twitter\':\'BordeauxJS\',\'website\':\'http:\\/\\/www.meetup.com\\/BordeauxJS\\/\'}',
    event_count: '7',
    recent_event_count: '0',
    created: '2015-01-24 13:35:25',
    modified: '2016-07-22 04:00:01',
    Contacts: {
      twitter: 'BordeauxJS',
      website: 'http://www.meetup.com/BordeauxJS/',
    },
  },
  Venue: {
    id: '1',
    edition_id: '9',
    name: 'Le Node',
    address: '12 Rue Des Faussets',
    zipcode: '33000',
    city: 'Bordeaux',
    country_code: 'FR',
    latitude: '44.840373',
    longitude: '-0.570311',
    event_count: '277',
    created: '2015-01-24 14:50:42',
    modified: '2015-02-15 17:26:20',
    oneliner: 'Le Node, 12 Rue Des Faussets, 33000 Bordeaux, FR',
  },
  Edition: {
    id: '9',
    name: 'Bordeaux',
    slug: 'bordeaux',
    event_count: '906',
    organization_count: '0',
    created: '2015-01-24 13:35:02',
    modified: '2015-01-24 13:35:02',
  },
};

export default Organization;
