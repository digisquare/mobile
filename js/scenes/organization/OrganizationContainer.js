import React, { Component, PropTypes } from 'react';
import { StyleSheet, Linking, TouchableOpacity } from 'react-native';
import { connect } from 'react-redux';
import Icon from 'react-native-vector-icons/FontAwesome';

import DigiColors from '../../common/DigiColors';

import Organization from './Organization';

const styles = StyleSheet.create({
  itemWrapper: {
    flex: 1,
    justifyContent: 'center',
    padding: 10,
  },
});

class OrganizationContainer extends Component {
  static route = {
    navigationBar: {
      title: 'Organisateur',
      renderRight: (route) => {
        const { params: { organization } } = route;
        return organization.Organization.Contacts.website && (
          <TouchableOpacity
            onPress={() => Linking.openURL(organization.Organization.Contacts.website)}
            style={styles.itemWrapper}
          >
            <Icon name="link" size={20} color={DigiColors.invertedFontColor} />
          </TouchableOpacity>
        );
      },
    },
  }

  render() {
    const { organization, organizations } = this.props;
    const { edition_id: editionId, venue_id: venueId } = organization.Organization;

    if (editionId && organizations[editionId]) {
      const organizationsItems = organizations[editionId].items;
      if (venueId && !organization.Venue && organizationsItems.length > 0) {
        const venue = organizationsItems.find(item => item.Venue.id === venueId);
        organization.Venue = venue ? venue.Venue : null; // eslint-disable-line no-param-reassign
      }
    }

    return (
      <Organization organization={organization} />
    );
  }
}

const mapStateToProps = state => ({
  organizations: state.organizations,
});

export default connect(mapStateToProps)(OrganizationContainer);

OrganizationContainer.propTypes = {
  organization: PropTypes.object.isRequired,
  organizations: PropTypes.object.isRequired,
};
