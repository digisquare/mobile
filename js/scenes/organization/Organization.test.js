import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';

import Organization from './Organization';

import organization from './Organization.mock';

it('renders correctly', () => {
  const tree = renderer.create(
    <Organization organization={organization} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
