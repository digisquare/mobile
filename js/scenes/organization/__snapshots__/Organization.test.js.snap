exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }>
  <ScrollView
    style={
      Object {
        "flex": 1,
      }
    }>
    <View
      style={
        Object {
          "padding": 10,
        }
      }>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "marginBottom": 20,
          }
        }>
        <View
          style={
            Object {
              "marginRight": 20,
            }
          }>
          <Image
            source={
              Object {
                "uri": "https://pbs.twimg.com/profile_images/2741451195/8ff14ab4849d749af06271795a15cf54_400x400.png",
              }
            }
            style={
              Object {
                "height": 80,
                "width": 80,
              }
            } />
        </View>
        <View
          style={undefined}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "black",
                "fontSize": 22,
                "fontWeight": "900",
                "marginTop": 10,
              }
            }>
            BordeauxJS
          </Text>
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "marginTop": 10,
              }
            }>
            <View
              style={
                Object {
                  "marginRight": 10,
                }
              }>
              <TouchableHighlight
                activeOpacity={0.85}
                onPress={[Function]}
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    Object {
                      "backgroundColor": "#55acee",
                      "borderRadius": 5,
                    },
                  ]
                }
                underlayColor="black">
                <View
                  padding={2}
                  paddingLeft={8}
                  paddingRight={8}
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "flex-start",
                        "padding": 8,
                      },
                      Object {
                        "backgroundColor": "#55acee",
                        "borderRadius": 5,
                      },
                      undefined,
                    ]
                  }>
                  <Text
                    accessible={true}
                    allowFontScaling={false}
                    ellipsizeMode="tail"
                    onPress={[Function]}
                    style={
                      Array [
                        Object {
                          "color": "white",
                          "fontSize": 12,
                        },
                        Object {
                          "marginRight": 10,
                        },
                        Object {
                          "fontFamily": "FontAwesome",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ]
                    }>
                    
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "fontWeight": "600",
                        },
                        Object {
                          "color": "white",
                        },
                      ]
                    }>
                    BordeauxJS
                  </Text>
                </View>
              </TouchableHighlight>
            </View>
            <View />
          </View>
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail">
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={null}>
          User group JavaScript à #bordeaux : réunions, discussions et live coding entre gens en asynchrone. #nodejs #coffeescript #javascript
        </Text>
      </Text>
    </View>
  </ScrollView>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "flexDirection": "row",
      }
    }>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "width": 60,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }>
        
      </Text>
    </View>
    <View
      style={
        Object {
          "paddingBottom": 5,
          "paddingLeft": 10,
          "paddingTop": 5,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontWeight": "500",
          }
        }>
        Le Node
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }>
        12 Rue Des Faussets
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }>
        33000 Bordeaux
      </Text>
    </View>
  </TouchableOpacity>
</View>
`;
